components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -212,6 +215,108 @@ time_t GetCurrentTimestamp() {
   return base::Time::NowFromSystemTime().ToTimeT();
 }
 
+// TODO(erogul): Remove these private observers and related classes when
+// GetCurrentBalanceReport and GetPublisherActivity are implemented via
+// callbacks
+class RewardsServicePrivateObserver : public base::CheckedObserver {
+ public:
+  ~RewardsServicePrivateObserver() override {}
+
+  virtual void OnGetCurrentBalanceReport(RewardsService* rewards_service,
+                                         const BalanceReport& balance_report) {}
+  virtual void OnGetPublisherActivityFromUrl(
+      RewardsService* rewards_service,
+      int error_code,
+      std::unique_ptr<ledger::PublisherInfo> info,
+      uint64_t windowId) {}
+};
+
+base::ObserverList<RewardsServicePrivateObserver> private_observers;
+
+#if BUILDFLAG(ENABLE_EXTENSIONS)
+class ExtensionRewardsServicePrivateObserver
```

Comment:
we don't want a separate implementation here, the existing ExtensionRewardsServiceObserver should implement this. You can make it a `friend` class